x-pack/legacy/plugins/epm/server/datasources/create.ts

```diff
@@ -8,76 +8,94 @@ import { SavedObjectsClientContract } from 'src/core/server/';
 import { CallESAsCurrentUser } from '../lib/cluster_access';
 import { installPipelines } from '../lib/elasticsearch/ingest_pipeline/ingest_pipelines';
 import { installTemplates } from '../lib/elasticsearch/template/install';
-import { AssetReference } from '../../common/types';
+import { AssetReference, InstallationStatus, RegistryPackage } from '../../common/types';
 import { SAVED_OBJECT_TYPE_DATASOURCES } from '../../common/constants';
 import { Datasource, Asset, InputType } from '../../../ingest/server/libs/types';
+import { getPackageInfo } from '../packages';
 import * as Registry from '../registry';
 
+const pkgToPkgKey = ({ name, version }: RegistryPackage) => `${name}-${version}`;
```

Comment:
General question not related to this PR: I would normally put a method which is running with `RegistryPackage` close to the definition of it so it is easy to find again for others. I'm aware this one is only used here, but I'm pretty sure it will be used in other places in the future. In Typescript I see this pattern of the `types.ts` file which does not contain any functions. Is this the pattern to follow? Just curious.